src/mongoose_client_api/mongoose_client_api_messages.erl

```diff
@@ -128,7 +128,14 @@ encode(Msg, Timestamp) ->
                                         <<"http://www.jivesoftware.com/xmlns/xmpp/properties">>),
 
     BodyTag = exml_query:path(Msg, [{element, <<"body">>}]),
-
+    Thread = exml_query:path(Msg, [{element, <<"thread">>}, cdata]), 
+    ThreadParentList =
+      case exml_query:path(Msg, [{element, <<"thread">>}, {attr, <<"parent">>}], undefined) of
```

Comment:
No make sense put default value `undefined` cuz by default used default value `undefined` - [exml_query.erl#L64](https://github.com/esl/exml/blob/master/src/exml_query.erl#L64).

Reply:
Okay, will remove the default case, replace with the improved code you made above.